.github/workflows/scripts/kinds/verify-kinds.go

```diff
@@ -0,0 +1,414 @@
+)
+
+const (
+	GITHUB_OWNER = "grafana"
+	GITHUB_REPO  = "kind-registry"
+)
+
+// main This script verifies that stable kinds are not updated once published (new schemas
+// can be added but existing ones cannot be updated).
+// If the env variable CODEGEN_VERIFY is not present, this also generates kind files into a
+// local "next" folder, ready to be published in the kind-registry repo.
+// If kind names are given as parameters, the script will make the above actions only for the
+// given kinds.
+func main() {
+	var kindArgs []string
+	if len(os.Args) > 1 {
+		kindArgs = os.Args[1:]
+	}
+
+	var corek []kindsys.Kind
+	var compok []kindsys.Composable
+
+	// This script will reach the GH API rate limit if ran for all kinds without token.
+	// If you don't have a GH token, run the script with kind names as parameters to run
+	// it only for a limited set of kinds.
+	var ts oauth2.TokenSource
+	token, ok := os.LookupEnv("GITHUB_TOKEN")
+	if ok {
+		ts = oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: token},
+		)
+	}
+
+	ctx := context.Background()
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+
+	// Search for the latest version directory present in the kind-registry repo
+	latestRegistryDir, err := findLatestDir(ctx, client)
+	if err != nil {
+		die(fmt.Errorf("failed to get latest directory for published kinds: %s", err))
+	}
+
+	errs := make([]error, 0)
+
+	// Kind verification
+	for _, kind := range corekind.NewBase(nil).All() {
+		if len(kindArgs) > 0 && !contains(kindArgs, kind.Name()) {
+			continue
+		}
+
+		name := kind.Props().Common().MachineName
+		err := verifyKind(kind, name, "core", latestRegistryDir)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		corek = append(corek, kind)
+	}
+
+	for _, pp := range corelist.New(nil) {
+		// ElasticSearch composable kind causes the CUE evaluator to hand
+		// see https://github.com/grafana/grafana/pull/68034#discussion_r1187800059
+		if pp.Properties.Id == "elasticsearch" {
+			continue
+		}
+
+		for _, kind := range pp.ComposableKinds {
+			if len(kindArgs) > 0 && !contains(kindArgs, kind.Name()) {
+				continue
+			}
+
+			si, err := kindsys.FindSchemaInterface(kind.Def().Properties.SchemaInterface)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+
+			name := strings.ToLower(fmt.Sprintf("%s/%s", strings.TrimSuffix(kind.Lineage().Name(), si.Name()), si.Name()))
+			err = verifyKind(kind, name, "composable", latestRegistryDir)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+
+			compok = append(compok, kind)
+		}
+	}
+
+	die(errs...)
+
+	if _, set := os.LookupEnv("CODEGEN_VERIFY"); set {
+		os.Exit(0)
+	}
+
+	// File generation
+	jfs := codejen.NewFS()
+	registryPath := filepath.Join(".github", "workflows", "scripts", "kinds")
+
+	coreJennies := codejen.JennyList[kindsys.Kind]{}
+	coreJennies.Append(
+		KindRegistryJenny(registryPath, kindArgs),
+	)
+	corefs, err := coreJennies.GenerateFS(corek...)
```

Comment:
More of an FYI note - if there's just one jenny actually going to be used here (the `KindRegistryJenny`), it's not actually necessary to create a `JennyList`.

It's fine, even preferable to keep this as-is given that we do it everywhere else, and until we make some improvements to the jenny library it's less familiar to do things that way